include/defines.hpp

```diff
@@ -250,4 +227,15 @@ enum EncodingMode
     GZIP = 0b00000010,
 };
 
+enum EventProcessingState
+{
+    INITIAL                 = 0b00000000,
+    PROCESSING              = 0b00001000,
+    WRITE_CHUNKED           = 0b00000100,
```

Comment:
the order here is intentional?

Reply:
Since uesd the bit map for the bit operations the order doesn't has meaning anymore only matters for readability concern.